packages/zapp/console/src/components/Launch/LaunchForm/inputHelpers/union.ts

```diff
@@ -0,0 +1,62 @@
+import { Core } from 'flyteidl';
+import { isObject } from 'lodash';
+import { InputTypeDefinition, InputValue, UnionValue } from '../types';
+import { getHelperForInput } from './getHelperForInput';
+import { ConverterInput, InputHelper, InputValidatorParams } from './types';
+import t from '../../../common/strings';
+
+function fromLiteral(literal: Core.ILiteral, inputTypeDefinition: InputTypeDefinition): InputValue {
+  const { listOfSubTypes } = inputTypeDefinition;
+  if (!listOfSubTypes?.length) {
+    throw new Error(t('missingUnionListOfSubType'));
+  }
+
+  // loop though the subtypes to find the correct match literal type
+  for (let i = 0; i < listOfSubTypes.length; i++) {
+    try {
+      const value = getHelperForInput(listOfSubTypes[i].type).fromLiteral(
+        literal,
+        listOfSubTypes[i],
+      );
+      return { value, typeDefinition: listOfSubTypes[i] } as UnionValue;
+    } catch (error) {
+      // do nothing here it's expected to have error
```

Comment:
nit: If you know which exactly error is expected to be here - can you please add it to the comment, so if in future we will receive a different error we will be aware that it wasn't expected by design

Reply:
added some detail !